contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,174 @@ doc: |
-#          one refers to.
-#     </doc>
-#     <group type="NXcoordinate_system"/>
-#     <!--How to interpret "depends_on" calls for NXtransformations instances which live
-# within an instance of NXcoordinate_system(_set)?
-# Depends on cardinality of NXcoordinate_system_set, i.e. how many NXcoordinate_system are defined?
-# - 0, the root of an NXtransformation chain i.e. "." means McStas
-# - 1, the root of an NXtransformation chain i.e. "." means "that specific CS" i.e. the one defined by the single instance of NXcoordinate_system in the set
-# - > 1, "." becomes ambiguous. My suggestion, spell out the name of the specific NXcoordinate_system instance to be used, using "."
-# despite all these, a warning should be raised, like a logic warning (and McStas kick in) or an error (i.e. be forbidden) be raised-->
-# </definition>
+  # frequently used coordinate systems with frequently relevant specializations follow
+  # convention 1 of DOI: 10.1088/0965-0393/23/8/083501 is implemented by inheriting type from NXcoordinate_system
+  processing_reference_frame(NXcoordinate_system):
+    doc: |
+      Details about eventually relevant named directions that may give reasons for anisotropies.
+      The classical example is mechanical processing where one has to specify which directions
+      (e.g. rolling, transverse, and normal direction) align how with the direction of the base
+      vectors of the sample_reference_frame.
+      
+      Reference DOI: 10.1016/j.matchar.2016.04.008 suggest to label the
+      base vectors of this coordinate system as Xp, Yp, Zp.
+    origin(NX_CHAR):  # specialized the inherited concept origin from NXcoordinate_system
+      doc: |
+        Location of the origin of the processing_reference_frame.
+        
+        We assume regions-of-interest in this reference frame form a
+        rectangle or cuboid. Edges are interpreted by inspecting the
+        direction of their outer unit normals (which point either parallel
+        or antiparallel) along respective base vector direction of the
+        reference frame.
+      enumeration: [undefined, front_top_left, front_top_right, front_bottom_right, front_bottom_left, back_top_left, back_top_right, back_bottom_right, back_bottom_left]
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of the processing_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
+        If a detector is involved, we assume the configuration is inspected from a position
+        that is located behind that detector.
+      enumeration: [undefined, north, east, south, west, in, out]
+    y_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing y-axis base vector of the processing_reference_frame.
+        We assume the configuration is inspected by looking towards the sample surface.
```

Comment:
I think these assumptions are okay, but I would still allow users to define their own assumptions.

How about
```yaml
It is assumed the configuration is inspected by looking towards the sample surface. If a detector is involved, it is assumed the configuration is inspected from a position that is located behind that detector.

If any of these assumptions is not met, the user is asked to explicitly state this.
```
And then simply add these kinds of sentences everywhere where there is an assumption taken.

Alternatively, you could add such a statement it in the doc of the class itself, but I would prefer to have it with the individual concepts.

Reply:
Agree that it should be with the concepts individually, enabling additional assumptions I support as the base class should offer service beyond materials engineering/condensed matter physics but then one cannot ask the user needs to make this explicit by overwriting the docstring of the concept, i.e.

I implemented:

If any of these assumptions is not met, the user is required to explicitly state this.